src/coreclr/vm/clsload.cpp

```diff
@@ -3461,6 +3461,12 @@ VOID ClassLoader::AddAvailableClassHaveLock(
     if (SUCCEEDED(pMDImport->GetNestedClassProps(classdef, &enclosing))) {
         // nested type
 
+        if (enclosing == TokenFromRid(mdtTypeDef, 1))
```

Comment:
Isn't it `TokenFromRid(1, mdtTypeDef)`?

Reply:
There appears to be an IsGlobalTypeDef helper for this

Reply:
I've updated the PR to fix this and the other issues where the TokenFromRid api is used with parameters in the wrong order. Fortunately, this shouldn't actually have any effect on the generated code, but it is kinda terrible.